<?xml version="1.0" encoding="UTF-8" ?>

<launch>
  <arg name="mode" default="$(env MOVEL_MODE)" doc="mode: [deployment, development]"/>
  <arg name="camera" default="camera"/>
  <arg name="two_phase" default="false"/>

  <group if="$(eval mode == 'deployment')">
    <node pkg="dalu_docking" name="movel_diff_drive_docking" type="diff_drive_docking" output="screen">
      <rosparam file="$(find dalu_docking)/config/diff_drive_docking_single_phase.yaml" command="load"/>
      <rosparam file="$(env HOME)/.config/movel/config/dalu_docking/config/diff_drive_docking_single_phase.yaml" command="load" />
      <param name="two_phase" type="bool" value="$(arg two_phase)"/>
    </node>
  
    <node pkg="dalu_docking" name="movel_tag_offset" type="tag_offset_plus" output="screen">
      <rosparam file="$(find dalu_docking)/config/tag_offset.yaml" command="load"/>
      <rosparam file="$(env HOME)/.config/movel/config/dalu_docking/config/tag_offset.yaml" command="load" />
    </node>
  
    <!-- April tag publisher -->
    <arg name="april_node_namespace" default="apriltag_ros_continuous_node" />
    <rosparam command="load" file="$(find dalu_docking)/config/april_settings.yaml" ns="$(arg april_node_namespace)" />
    <rosparam command="load" file="$(env HOME)/.config/movel/config/dalu_docking/config/april_settings.yaml" ns="$(arg april_node_namespace)" />
    <rosparam command="load" file="$(find dalu_docking)/config/april_tags.yaml" ns="$(arg april_node_namespace)" />
    <rosparam command="load" file="$(env HOME)/.config/movel/config/dalu_docking/config/april_tags.yaml" ns="$(arg april_node_namespace)" />
    <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg april_node_namespace)" clear_params="true" output="screen" >
      <remap from="image_rect" to="/$(arg camera)/color/image_raw" />
      <remap from="camera_info" to="/$(arg camera)/color/camera_info" />
      <param name="camera_frame" type="str" value="$(arg camera)_color_optical_frame" />
      <param name="publish_tag_detections_image" type="bool" value="false" /> 
    </node>
  </group>

  <group unless="$(eval mode == 'deployment')">
    <node pkg="dalu_docking" name="movel_diff_drive_docking" type="diff_drive_docking" output="screen">
      <rosparam file="$(find dalu_docking)/config/diff_drive_docking_single_phase.yaml" command="load"/>
      <param name="two_phase" type="bool" value="$(arg two_phase)"/>
    </node>
  
    <node pkg="dalu_docking" name="movel_tag_offset" type="tag_offset_plus" output="screen">
      <rosparam file="$(find dalu_docking)/config/tag_offset.yaml" command="load"/>
    </node>
  
    <!-- April tag publisher -->
    <arg name="april_node_namespace" default="apriltag_ros_continuous_node" />
    <rosparam command="load" file="$(find dalu_docking)/config/april_settings.yaml" ns="$(arg april_node_namespace)" />
    <rosparam command="load" file="$(find dalu_docking)/config/april_tags.yaml" ns="$(arg april_node_namespace)" />
    <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg april_node_namespace)" clear_params="true" output="screen" >
      <remap from="image_rect" to="/$(arg camera)/color/image_raw" />
      <remap from="camera_info" to="/$(arg camera)/color/camera_info" />
      <param name="camera_frame" type="str" value="$(arg camera)_color_optical_frame" />
      <param name="publish_tag_detections_image" type="bool" value="false" /> 
    </node>
  </group>
</launch>
